flink/src/test/java/org/apache/iceberg/flink/TestFlinkFilters.java

```diff
@@ -0,0 +1,361 @@
+import org.apache.flink.table.expressions.CallExpression;
+import org.apache.flink.table.expressions.Expression;
+import org.apache.flink.table.expressions.FieldReferenceExpression;
+import org.apache.flink.table.expressions.ResolvedExpression;
+import org.apache.flink.table.expressions.UnresolvedCallExpression;
+import org.apache.flink.table.expressions.UnresolvedReferenceExpression;
+import org.apache.flink.table.expressions.ValueLiteralExpression;
+import org.apache.flink.table.expressions.utils.ApiExpressionDefaultVisitor;
+import org.apache.flink.table.functions.BuiltInFunctionDefinitions;
+import org.apache.iceberg.expressions.And;
+import org.apache.iceberg.expressions.BoundLiteralPredicate;
+import org.apache.iceberg.expressions.Not;
+import org.apache.iceberg.expressions.Or;
+import org.apache.iceberg.expressions.UnboundPredicate;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.util.DateTimeUtil;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestFlinkFilters {
+
+  private static final TableSchema TABLE_SCHEMA = TableSchema.builder()
+      .field("field1", DataTypes.INT())
+      .field("field2", DataTypes.BIGINT())
+      .field("field3", DataTypes.FLOAT())
+      .field("field4", DataTypes.DOUBLE())
+      .field("field5", DataTypes.STRING())
+      .field("field6", DataTypes.BOOLEAN())
+      .field("field7", DataTypes.BINARY(2))
+      .field("field8", DataTypes.DECIMAL(10, 2))
+      .field("field9", DataTypes.DATE())
+      .field("field10", DataTypes.TIME())
+      .field("field11", DataTypes.TIMESTAMP())
+      .field("field12", DataTypes.TIMESTAMP_WITH_LOCAL_TIME_ZONE())
+      .build();
+
+  // A map list of fields and values used to verify the conversion of flink expression to iceberg expression
+  private static final List<Tuple2<String, Object>> FIELD_VALUE_LIST = ImmutableList.of(
+      Tuple2.of("field1", 1),
+      Tuple2.of("field2", 2L),
+      Tuple2.of("field3", 3F),
+      Tuple2.of("field4", 4D),
+      Tuple2.of("field5", "iceberg"),
+      Tuple2.of("field6", true),
+      Tuple2.of("field7", new byte[] {'a', 'b'}),
+      Tuple2.of("field8", BigDecimal.valueOf(10)),
+      Tuple2.of("field9", DateTimeUtil.daysFromDate(LocalDate.now())),
+      Tuple2.of("field10", DateTimeUtil.microsFromTime(LocalTime.now())),
+      Tuple2.of("field11", DateTimeUtil.microsFromTimestamp(LocalDateTime.now())),
+      Tuple2.of("field12", DateTimeUtil.microsFromInstant(Instant.now()))
+  );
+
+  @Test
+  public void testFlinkDataTypeEqual() {
+    matchLiteral("field1", 1, 1);
+    matchLiteral("field2", 10L, 10L);
+    matchLiteral("field3", 1.2F, 1.2F);
+    matchLiteral("field4", 3.4D, 3.4D);
+    matchLiteral("field5", "abcd", "abcd");
+    matchLiteral("field6", true, true);
+    matchLiteral("field7", new byte[] {'a', 'b'}, ByteBuffer.wrap(new byte[] {'a', 'b'}));
+    matchLiteral("field8", BigDecimal.valueOf(10), BigDecimal.valueOf(10));
```

Comment:
`valueOf` creates a decimal with a scale of 0, but the type's scale is 2. Can you create a big decimal that matches the type?